src/Ogooreck/API/ApiSpecification.cs

```diff
@@ -0,0 +1,174 @@
+﻿using System.Net;
+using System.Net.Http.Json;
+using FluentAssertions;
+using Microsoft.AspNetCore.Mvc.Testing;
+
+namespace Ogooreck.API;
+
+public static class ApiSpecification
+{
+    ///////////////////
+    ////   GIVEN   ////
+    ///////////////////
+    public static Func<HttpRequestMessage, HttpRequestMessage> URI(string uri) =>
+        URI(new Uri(uri, UriKind.RelativeOrAbsolute));
+
+    public static Func<HttpRequestMessage, HttpRequestMessage> URI(Uri uri) =>
+        request =>
+        {
+            request.RequestUri = uri;
+            return request;
+        };
+
+    public static Func<HttpRequestMessage, HttpRequestMessage> BODY<T>(T body) =>
+        request =>
+        {
+            request.Content = JsonContent.Create(body);
+
+            return request;
+        };
+
+    ///////////////////
+    ////   WHEN    ////
+    ///////////////////
+    public static Func<HttpClient, HttpRequestMessage, Task<HttpResponseMessage>> GET = SEND(HttpMethod.Post);
+
+    public static Func<HttpClient, HttpRequestMessage, Task<HttpResponseMessage>> POST = SEND(HttpMethod.Post);
+
+    public static Func<HttpClient, HttpRequestMessage, Task<HttpResponseMessage>> PUT = SEND(HttpMethod.Put);
+
+    public static Func<HttpClient, HttpRequestMessage, Task<HttpResponseMessage>> DELETE = SEND(HttpMethod.Delete);
+
+    public static Func<HttpClient, HttpRequestMessage, Task<HttpResponseMessage>> SEND(HttpMethod httpMethod) =>
+        (api, request) =>
+        {
+            request.Method = httpMethod;
+            return api.SendAsync(request);
+        };
+
+    ///////////////////
+    ////   THEN    ////
+    ///////////////////
+    public static Action<HttpResponseMessage> OK = HTTP_STATUS(HttpStatusCode.OK);
+
+    public static Action<HttpResponseMessage> CREATED =>
+        response =>
+        {
+            HTTP_STATUS(HttpStatusCode.Created);
+
+            var locationHeader = response.Headers.Location;
+
+            locationHeader.Should().NotBeNull();
+
+            var location = locationHeader!.OriginalString;
+
+            location.Should().StartWith(response.RequestMessage!.RequestUri!.AbsolutePath);
+        };
+
+    public static Action<HttpResponseMessage> BAD_REQUEST = HTTP_STATUS(HttpStatusCode.BadRequest);
+    public static Action<HttpResponseMessage> NOT_FOUND = HTTP_STATUS(HttpStatusCode.NotFound);
+    public static Action<HttpResponseMessage> CONFLICT = HTTP_STATUS(HttpStatusCode.Conflict);
+    public static Action<HttpResponseMessage> PRECONDITION_FAILED = HTTP_STATUS(HttpStatusCode.PreconditionFailed);
+    public static Action<HttpResponseMessage> METHOD_NOT_ALLOWED = HTTP_STATUS(HttpStatusCode.MethodNotAllowed);
+
+    public static Action<HttpResponseMessage> HTTP_STATUS(HttpStatusCode status) =>
```

Comment:
Personally, I'm not a fan of the "all caps" naming...would prefer camel-casing, which I believe is the Microsoft code-style standard for .Net

Reply:
Thank you for the suggestion. I'll keep that in mind, and gather also other feedback around that. I like that it makes tests more explicit and the specific parts highlighted, so I'll keep it as it is for now.